src/IndexingMsg.chpl

```diff
@@ -145,25 +186,63 @@ module IndexingMsg
     /* arrayViewIntIndexAssignMsg "av[int_list]=value" response to __getitem__(int_list) where av is an ArrayView */
     proc arrayViewIntIndexAssignMsg(cmd: string, payload: string, st: borrowed SymTab): MsgTuple throws {
         param pn = Reflection.getRoutineName();
-        var (pdaName, dtypeStr, dimProdName, coordsName, value) = payload.splitMsgToTuple(5);
+        var msgArgs = parseMessageArgs(payload, 5);
+        const pdaName = msgArgs.getValueOf("base");
+        const dimProdName = msgArgs.getValueOf("dim_prod");
+        const coordsName = msgArgs.getValueOf("coords");
+        const dtypeStr = msgArgs.getValueOf("dtype");
+        var value = msgArgs.getValueOf("value");
         imLogger.debug(getModuleName(),getRoutineName(),getLineNumber(), "%s %s %s %s".format(cmd, pdaName, dimProdName, coordsName));
 
         var dimProd: borrowed GenSymEntry = getGenericTypedArrayEntry(dimProdName, st);
         var dimProdEntry = toSymEntry(dimProd, int);
         var coords: borrowed GenSymEntry = getGenericTypedArrayEntry(coordsName, st);
 
+        // map used to generate the "array" key for intIndexMsg
+        var arrayMap = new map(string, string);
+        arrayMap.add("key", "array");
+        arrayMap.add("val", pdaName);
+        arrayMap.add("objType", "PDARRAY");
+        arrayMap.add("dtype", "");
+
+        // map used to generate the "dtypestr" key for intIndexMsg
+        //TODO - idk if we need this because the json format passes the dtype string
```

Comment:
did you determine if we still need `dtypeStr` or not?

Reply:
Yes we do. I forgot to remove the todo.